capella_model_explorer/reports.py

```diff
@@ -191,22 +212,31 @@ def _register_template_category(category: str) -> None:
 def _make_href(
     obj: m.ModelElement | m.AbstractDiagram,
 ) -> str | None:
+    generic_template: Template | None = None
     for template in state.templates:
-        if template.scope is not None:
-            clsname = template.scope.type
-            if obj.xtype.rsplit(":", 1)[-1] == clsname:
-                return app.app.url_path_for(
-                    "template_page",
-                    template_id=template.id,
-                    model_element_uuid=obj.uuid,
-                )
-        if "__generic__" in template.path.stem:
-            return app.app.url_path_for(
-                "template_page",
-                template_id=template.id,
-                model_element_uuid=obj.uuid,
-            )
-    return fh.RedirectResponse(app.app.url_path_for("home"))
+        if template.path.name.startswith("__generic__"):
+            generic_template = template
+            continue
+
+        if template.single:
+            continue
+
+        if template.scope and not template.scope.applies_to(obj):
+            continue
+
+        return app.app.url_path_for(
+            "template_page",
+            template_id=template.id,
+            model_element_uuid=obj.uuid,
+        )
```

Comment:
Can we express this more positive?
```suggestion
        if template.scope and template.scope.applies_to(obj):
            return app.app.url_path_for(
                "template_page",
                template_id=template.id,
                model_element_uuid=obj.uuid,
            )
```

Reply:
There aren't the same, the inversion would be "if **not** template.scope **or** template.scope.applies_to(obj)". And then you have a negative in it again 🙃 (Although it's questionable whether a template without a scope even makes sense in the first place - but that's something for a different PR, as it would also affect the `__generic__` template.)